insonmnia/miner/builder.go

```diff
@@ -75,11 +76,20 @@ func (b *MinerBuilder) Build() (miner *Miner, err error) {
 	}
 
 	if b.ip == nil {
-		addr, err := util.GetPublicIP()
+		log.G(b.ctx).Debug("discovering public IP address with NAT type, this may take a long ...")
+
+		client := stun.NewClient()
```

Comment:
Does it use some default STUN server? Maybe we need to make it also configurable